apps/emqx_eviction_agent/src/emqx_eviction_agent_channel.erl

```diff
@@ -221,9 +223,9 @@ set_expiry_timer(#{conninfo := ConnInfo} = Channel) ->
             {error, should_be_expired}
     end.
 
-open_session(ConnInfo, #{clientid := ClientId} = ClientInfo) ->
+open_session(ConnInfo, #{clientid := ClientId} = ClientInfo, MaybeWillMsg) ->
     Channel = channel(ConnInfo, ClientInfo),
-    case emqx_cm:open_session(_CleanSession = false, ClientInfo, ConnInfo) of
+    case emqx_cm:open_session(_CleanSession = false, ClientInfo, ConnInfo, MaybeWillMsg) of
```

Comment:
This call should check the BPAPI version of the callee, since eviction is likely to happen during upgrade.